characters_counter/characters_counter.py

```diff
@@ -1,16 +1,56 @@
-from collections import Counter
+import collections
+import re
+from collections import Counter, defaultdict
 from functools import cache
+from pathlib import Path
 
 REQUIREMENT_OCCURRENCE = 1
+NUM_OF_MOST_COMMON_CHARACTERS = 3
 
 
 @cache
 def count_characters_occurrence(string: str) -> int:
     """
-    The function returns the number of characters in the string occuring only
-    once.
+    The function returns the number of characters in the string occuring only once.
     """
     characters_counts = Counter(string)
     unique_occurrences = filter(lambda occurrence: (occurrence == REQUIREMENT_OCCURRENCE), characters_counts.values())
     unique_characters_number = sum(unique_occurrences)
     return unique_characters_number
+
+
+def get_frequent_characters_per_sentence(sentence: str) -> dict:
+    """
+    The helper function counts the frequency of each character in the input sentence, excluding spaces.
+    Single word is considered as a sentence as well.
+    The number of characters returned is controlled by the global constant "NUM_OF_MOST_COMMON_CHARACTERS".
+    :param sentence: str
+    :return: dict[str, int]
+    """
+    count_characters = Counter(sentence.replace(" ", ""))
+    return dict(count_characters.most_common(NUM_OF_MOST_COMMON_CHARACTERS))
+
+
+def count_character_frequency(data_input: Path) -> defaultdict[str, Counter[str]] | dict[str, int] | None:
+    """
+    The function counts most used letters/symbols in each sentence from the given textfile.
+    :param data_input: Path
+    :return: dict[str, int]
+    """
+    if Path(data_input).suffix:
```

Comment:
Separate function please.

Reply:
I hope, I undestood you correctly this time:
I separated my code. Now I have:
- a function that attemps to open and read file. If fhe file is not found, the exception is raised;
- a function that makes a lot of operations with a text. (I unite all operations but I have doubs if it is right);
- a function that accepts data input and return result.